crates/bevy_asset/src/io/metadata.rs

```diff
@@ -0,0 +1,58 @@
+/// A enum representing a type of file.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum FileType {
+    Directory,
+    File,
+    // To be compatible with [`std::fs::FileType`].
+    Symlink,
```

Comment:
I don't think we should expose the case of SymLink. What happens if a user tries to load an asset that's a symlink?

If the link is followed and the asset loaded, we should show the filetype of the link target. If it's not, we should fail in a similar way (or fix link handling)

Reply:
I tested `std::fs::metadata` with symlinks on linux and windows, `is_symlink()` was never true but `is_dir()` or `is_file()`. And the links were followed.

Reply:
Yep. `fs::metadata` will follow symlinks, but there is a `fs::symlink_metadata` which does not :)